### PR TITLE
Check if diagnostic file is in relative path after absolute

### LINKED ISSUE #93: `--diagnostic-path` needs to include directory
```
$ ./frame --diagnostic-path foo
# ...
std::exception::what: Diagnostic path () does not exist

$ touch foo
$ ./frame --diagnostic-path foo
# ...
std::exception::what: Diagnostic path () does not exist

$ ./frame --diagnostic-path $PWD/foo
# _now_ it's happy
```

Help doesn't say it needs to be absolute:
```
  --diagnostic-path arg                 Path (including filename) of diagnostic
                                        file
```

### PR DESCRIPTION
**Pull request by graysonguarino**
~~Fixes #93 by clarifying an absolute path is needed for the `--diagnostic-path` option.~~

~~This is if we consider `./` to be a macro that gets an absolute path from a relative path. Not sure how to specify this without getting wordier.~~

Fixes #93 by first checking if file is in absolute path, then attempts relative path if that fails.

### CODE REVIEW

**Review by Saviq — CHANGES_REQUESTED** [review]
As [Alan commented](https://github.com/MirServer/ubuntu-frame/issues/93#issuecomment-1287182956), it's not actually that, rather it needs to resolve a relative path without slashes?

Not sure where this (weird) behaviour comes from…

File: src/frame_main.cpp [hunk 1]
```diff
@@ -55,7 +55,7 @@ int main(int argc, char const* argv[])
             ConfigurationOption{[&](auto& option) { background_client.set_crash_text_colour(option);},
                                "diagnostic-text",       "Colour of diagnostic screen text RGB", "0xffffff"},
             ConfigurationOption{[&] (auto& option) { background_client.set_diagnostic_path(option);},
-                               "diagnostic-path",  "Path (including filename) of diagnostic file", ""},
+                               "diagnostic-path",  "Absolute path (including filename) of diagnostic file", ""},
```
Comment (AlanGriffiths) [inline src/frame_main.cpp:58]: This is wrong. Relative paths also work

**Review by AlanGriffiths — CHANGES_REQUESTED** [review]
The problem is that `diagnostic-path` is split into "directory" and "file" and then existence of the directory is tested. But, if only a filename is supplied then "directory" is empty and the logic breaks.

**Comment by graysonguarino** [comment]
It may be wise to drop this PR in favor of merging only #97. They both dealt with the same lines of the same code and the newer PR fixes both #93 and #94.

**Pull request closed without merging**